django/db/models/query.py

```diff
@@ -470,6 +470,41 @@ def bulk_create(self, objs, batch_size=None):
 
         return objs
 
+    def bulk_save(self, objs, fields, batch_size=None):
+        """
+        Update the given fields in each of the given objects in the database.
+        Do *not* call save() on each of the instances.
+        """
+        assert batch_size is None or batch_size > 0
+        if not fields:
+            raise ValueError('Specific field names must be given to bulk_save')
+        objs = list(objs)
```

Comment:
I have a concern with forcing the QuerySet to evaluate here. If there are a very large amount of records (millions), this could be detrimental to the memory of whatever system it is running on because it will [populate the model's cache](https://docs.djangoproject.com/en/2.0/topics/db/queries/#caching-and-querysets). Is there not a way that we can support iterators here? 

I understand that this reduces the amount of time it takes, but we also have to be mindful of why someone such as myself would use it; which is that I have a ton of data that I need to update efficiently. The below is not the greatest example, but it lends itself to what @freakboy3742 mentioned in his [comment on the ticket](https://code.djangoproject.com/ticket/23646#comment:2).

```
queryset = Books.objects.values("pk").annotate(price=models.Sum("fk__value")).values("pk", "price").iterator()
# [{'pk': 1, 'price': 10}, {'pk': 2, 'price': 25}]
Books.objects.bulk_save(values=queryset)
```

Reply:
Hey Michael,
Thanks for your feedback, but there seems to be some confusion. In this case objs is an iterable of model instances, not a queryset. The API is very similar to bulk create, and in fact so is the code. The list cast is just to ensure that if someone passes in a generator we can use len() on it to create the batches.

Reply:
Oh yes, there definitely was some confusion. When looking at the code, I saw references to `obj.pk` and thought we were potentially dealing with a QuerySet (which I guess that _technically_ we _could_ be). My apologies! Carry on. 😋